File: backend/src/main/java/com/society/backend/entity/Vendor.java
package com.society.backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "vendors")
@Getter
@Setter
@NoArgsConstructor
public class Vendor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "society_id")
    private Society society; // null means common/partner vendor

    @Column(nullable = false)
    private String name;

    @Column(name = "service_type", nullable = false)
    private String serviceType; // PLUMBER, ELECTRICIAN, SECURITY, HOUSEKEEPING, etc.

    @Column(name = "contact_person")
    private String contactPerson;

    @Column(name = "contact_person_phone")
    private String contactPersonPhone;

    @Column(name = "contact_person_email")
    private String contactPersonEmail;

    @Column
    private String phone;

    @Column
    private String email;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Column(name = "gst_number")
    private String gstNumber;

    @Column(name = "pan_number")
    private String panNumber;

    @Column(name = "bank_name")
    private String bankName;

    @Column(name = "account_number")
    private String accountNumber;

    @Column(name = "ifsc_code")
    private String ifscCode;

    @Column(name = "is_common")
    private Boolean isCommon = false; // true = available to all societies

    @Column(name = "approval_status")
    private String approvalStatus = "PENDING"; // PENDING, APPROVED, REJECTED

    @Column(name = "is_active")
    private Boolean isActive = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: ios-app/SocietyManager/Models/Vendor.swift
import Foundation

struct Vendor: Codable, Identifiable {
    let id: Int
    let name: String
    let contactPerson: String?
    let contactPersonPhone: String?
    let contactPersonEmail: String?
    let phone: String?
    let email: String?
    let address: String?
    let gstNumber: String?
    let panNumber: String?
    let bankName: String?
    let accountNumber: String?
    let ifscCode: String?
    let serviceType: String?
    let societyId: Int?
    let societyName: String?
    let approvalStatus: String?
    let createdByUserId: Int?
    let createdByRole: String?
    let isActive: Bool?
    let createdAt: String?
}

struct VendorRequest: Encodable {
    let name: String
    let contactPerson: String?
    let contactPersonPhone: String?
    let contactPersonEmail: String?
    let phone: String?
    let email: String?
    let address: String?
    let gstNumber: String?
    let panNumber: String?
    let serviceType: String?
    let societyId: Int?
}
